info.limpet.test/src/info/limpet/data/TestOperations.java

```diff
@@ -800,6 +803,39 @@ public void testSubtraction()
   }
 
   @SuppressWarnings("unchecked")
+  public void testAddLayerOperation()
+  {
+    // place to store results data
+    InMemoryStore store = new SampleData().getData(10);
+
+    List<IStoreItem> selection = new ArrayList<IStoreItem>();
+
+    StoreGroup track1 = new StoreGroup("Track 1");
+    selection.add(track1);
+
+    Collection<ICommand<IStoreItem>> commands =
+        new AddLayerOperation().actionsFor(selection, store, context);
+    assertEquals("invalid number of inputs", 1, commands.size());
+    for (ICommand<IStoreItem> iCommand : commands)
+    {
+      iCommand.execute();
+    }
+
```

Comment:
the above command creates a new layer.  I think you should test that a layer actually got added.